benchmarks/modules/utils.py

```diff
@@ -267,11 +271,38 @@ def setup_spack_environment(self):
             f'(cd {cp_dir}; find . \( -name "spack.yaml" -o -name "compilers.yaml" -o -name "packages.yaml" \) -print0 | xargs -0 tar cf - | tar -C {dest} -xvf -)',
             f'spack -e {self.build_system.environment} config add "config:install_tree:root:{env_dir}/opt"',
         ]
-
+        # else f"{d[key].value}" if isinstance(d[key].value, bool) for boolean with "True" instead of True
+        cmd_spack_variants = 'from spack import environment; d = environment.active_environment().spec_lists["specs"].specs[0].variants.dict; keys = list(d.keys());values = {key: ([d[key].value[i] for i in range(len(d[key].value))] if isinstance(d[key].value, tuple) else d[key].value) for key in keys};print(values)'
```

Comment:
In all these commands, you extract only the first package from the spec with `print(environment.active_environment().spec_lists["specs"].specs[0]` . There should be a dictionary _for each package_ instead.

Reply:
I think it's not multiple specs, but multiple packages inside the one spec, see eg. https://github.com/ukri-excalibur/excalibur-tests/blob/0a60516fcc73c8914984bc7e36fbd739bb0caea2/benchmarks/apps/hpl/hpl.py#L17 
Do I get this right @giordano?

Reply:
> Do I get this right @giordano?

Almost, they are two distinct top-level specs in one environment.  But the point is that, as I had already explained in the second point of the caveats section of https://github.com/ukri-excalibur/excalibur-tests/issues/251#issuecomment-1914696265, you can't assume there's exactly 1 top-level spec in the environment, which is what you're doing when you do `specs[0]`.